container/lxd/lxd_test.go

```diff
@@ -201,3 +203,39 @@ func (t *LxdSuite) TestNetworkDevicesWithParentDevice(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, jc.DeepEquals, expected)
 }
+
+func (t *LxdSuite) TestGetImageSourcesDefaultConfig(c *gc.C) {
+	mgr := t.makeManager(c, t.baseConfig())
+
+	sources, err := lxd.GetImageSources(mgr)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(sources, gc.DeepEquals, lxdclient.DefaultImageSources)
+}
+
+func (t *LxdSuite) TestGetImageSourcesDefaultNonStandardStreamDefaultConfig(c *gc.C) {
+	cfg := t.baseConfig()
+	cfg[config.ContainerImageStreamKey] = "nope"
+	mgr := t.makeManager(c, cfg)
+
+	sources, err := lxd.GetImageSources(mgr)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(sources, gc.DeepEquals, lxdclient.DefaultImageSources)
+}
+
+func (t *LxdSuite) TestGetImageSourcesImageMetadataURLExpectedHTTPSSources(c *gc.C) {
+	cfg := t.baseConfig()
+	cfg[config.ContainerImageMetadataURLKey] = "http://special.container.sauce"
+	mgr := t.makeManager(c, cfg)
+
+	sources, err := lxd.GetImageSources(mgr)
+	c.Assert(err, jc.ErrorIsNil)
+
+	expectedSources := append([]lxdclient.Remote{{
+		Name:          "special.container.sauce",
+		Host:          "https://special.container.sauce",
+		Protocol:      lxdclient.SimplestreamsProtocol,
+		Cert:          nil,
+		ServerPEMCert: "",
+	}}, lxdclient.DefaultImageSources...)
+	c.Check(sources, gc.DeepEquals, expectedSources)
```

Comment:
If we are going to go down the route of "daily" => don't include released, then using a custom URL + image-stream=daily, should probably *not* include the "released" stream.

Reply:
Implemented this.